### ISSUE TITLE
panic when referencing unknown subpackage

### ISSUE THREAD

**Issue by borchero** [prefix-dev/rattler-build#1244]
Using `rattler-build` from `main` on darwin-aarch64, the following recipe where `test1` is referenced but not built

```yaml
recipe:
  name: test-split
  version: v1

build:
  number: 0
  noarch: python

source:
  path: ./

outputs:
  - package:
      name: test.a
  - package:
      name: test.b
    requirements:
      run:
        - ${{ pin_subpackage('test1', exact=True) }} # this package does not exist!
```

results in a panic:

```
thread 'main' panicked at src/variant_render.rs:178:55:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by wolfv** [comment]
Great, thanks for testing. This should definitely result in an error with a helpful message! Thanks.

**Comment by wolfv** [comment]
Thanks! Fixed and added a test in #1250
